include/nanobind/eigen/sparse.h

```diff
@@ -146,15 +146,116 @@ template <typename T> struct type_caster<T, enable_if_t<is_eigen_sparse_matrix_v
 /// Caster for Eigen::Map<Eigen::SparseMatrix>, still needs to be implemented.
 template <typename T>
 struct type_caster<Eigen::Map<T>, enable_if_t<is_eigen_sparse_matrix_v<T>>> {
+    using Scalar = typename T::Scalar;
+    using StorageIndex = typename T::StorageIndex;
+    using Index = typename T::Index;
+    using SparseMap = Eigen::Map<T>;
     using Map = Eigen::Map<T>;
     using SparseMatrixCaster = type_caster<T>;
-    static constexpr auto Name = SparseMatrixCaster::Name;
+    static constexpr bool RowMajor = T::IsRowMajor;
+
+    using ScalarNDArray = ndarray<numpy, Scalar, shape<-1>>;
+    using StorageIndexNDArray = ndarray<numpy, StorageIndex, shape<-1>>;
+
+    using ScalarCaster = make_caster<ScalarNDArray>;
+    using StorageIndexCaster = make_caster<StorageIndexNDArray>;
+
+    static constexpr auto Name = const_name<RowMajor>("scipy.sparse.csr_matrix[",
+                                           "scipy.sparse.csc_matrix[")
+                   + make_caster<Scalar>::Name + const_name("]");
+
     template <typename T_> using Cast = Map;
     template <typename T_> static constexpr bool can_cast() { return true; }
 
-    bool from_python(handle src, uint8_t flags, cleanup_list *cleanup) noexcept = delete;
+    ScalarCaster data_caster;
+    StorageIndexCaster indices_caster, indptr_caster;
+    Index rows, cols, nnz;
+
+    bool from_python(handle src, uint8_t flags, cleanup_list *cleanup) noexcept 
+    {
+        object obj = borrow(src);
+        try {
+            object matrix_type = module_::import_("scipy.sparse").attr(RowMajor ? "csr_matrix" : "csc_matrix");
+            if (!obj.type().is(matrix_type))
+                obj = matrix_type(obj);
+        } catch (const python_error &) {
```

Comment:
Sorry, another minor comment on the above. If it is the case that you do end up returning false if `!obj.type().is(matrix_type)` then the catch block of the exception I'd guess should probably re-throw the exception since it probably means something is wrong other than that the object is not a sparse matrix (e.g., an import or some other error). I.e., the return of true/false should probably (unless I don't understand things) only happen if there are no errors.  Similarly, in the below, when you are returning false, this means that it passes the above type check but there is still a problem getting, say, indices, and I'd guess that would mean something is quite wrong (not well indicated by a false return) with the object and/or setup.